pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -0,0 +1,342 @@
+package bootstrappolicy
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+)
+
+const (
+	InfraBuildControllerServiceAccountName = "build-controller"
+	BuildControllerRoleName                = "system:build-controller"
+
+	InfraReplicationControllerServiceAccountName = "replication-controller"
+	ReplicationControllerRoleName                = "system:replication-controller"
+
+	InfraDeploymentControllerServiceAccountName = "deployment-controller"
+	DeploymentControllerRoleName                = "system:deployment-controller"
+
+	InfraJobControllerServiceAccountName = "job-controller"
+	JobControllerRoleName                = "system:job-controller"
+
+	InfraHPAControllerServiceAccountName = "hpa-controller"
+	HPAControllerRoleName                = "system:hpa-controller"
+
+	InfraPersistentVolumeControllerServiceAccountName = "pv-controller"
+	PersistentVolumeControllerRoleName                = "system:pv-controller"
+)
+
+type InfraServiceAccounts struct {
+	serviceAccounts sets.String
+	saToRole        map[string]authorizationapi.ClusterRole
+}
+
+var InfraSAs = &InfraServiceAccounts{}
+
+func (r *InfraServiceAccounts) addServiceAccount(saName string, role authorizationapi.ClusterRole) error {
+	if _, exists := r.serviceAccounts[saName]; exists {
+		return fmt.Errorf("%s already registered", saName)
+	}
+
+	r.saToRole[saName] = role
+	r.serviceAccounts.Insert(saName)
+	return nil
+}
+
+func (r *InfraServiceAccounts) GetServiceAccounts() []string {
+	return r.serviceAccounts.List()
+}
+
+func (r *InfraServiceAccounts) RoleFor(saName string) (authorizationapi.ClusterRole, bool) {
+	ret, exists := r.saToRole[saName]
+	return ret, exists
+}
+
+func (r *InfraServiceAccounts) AllRoles() []authorizationapi.ClusterRole {
+	ret := []authorizationapi.ClusterRole{}
+	for _, role := range r.saToRole {
```

Comment:
> make sure we don't return duplicate roles

`addServiceAccount` now enforces it.  If someone bypasses that and directly adds to a "private(ish)" member, I think they get what they deserve.  I don't like enforcing data coherence in a `get`.